src/cascadia/TerminalControl/ThreadSafeOptional.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+#include "pch.h"
+
+template<typename T>
+class ThreadSafeOptional
+{
+public:
+    template<class... Args>
+    bool Emplace(Args&&... args)
+    {
+        std::lock_guard guard{ _lock };
+
+        bool hadValue = _inner.has_value();
+        _inner.emplace(std::forward<Args>(args)...);

```

Comment:
Oh ooops. My `ValueWatcher` didn't update it's value if it was already occupied. 😄